### PR TITLE
#3121 Manager API adds definition support

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3121 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/api/body/connectorBody.js [hunk 1]
```diff
@@ -14,36 +14,12 @@
...
-    workerClusterKey,
-    tags,
-  };
+export const request = params => {
+  const definitions = getConnect(params);
+  const body = createBody(definitions);
+  return body;
 };
 
 export const response = () => {
```
Comment (saivirtue) [inline ohara-manager/client/src/api/body/connectorBody.js]: we can remove settings body here also :)

File: ohara-manager/client/src/api/body/brokerBody.js [hunk 2]
```diff
@@ -14,62 +14,22 @@
...
-    exporterPort,
-    jmxPort,
-    zookeeperClusterKey,
-    nodeNames,
-    tags,
-  };
+export const request = params => {
+  const definitions = getCluster(params);
+  const body = createBody(definitions);
+  body['nodeNames'] = [array];
```
Comment (saivirtue) [inline ohara-manager/client/src/api/body/brokerBody.js]: redundant code?

File: ohara-manager/client/src/api/body/topicBody.js [hunk 3]
```diff
@@ -14,33 +14,12 @@
...
-    numberOfPartitions,
-    tags,
-  };
+export const request = params => {
+  const definitions = getTopic(params);
+  const body = createBody(definitions);
+  return body;
 };
 
 export const response = () => {
```
Comment (saivirtue) [inline ohara-manager/client/src/api/body/topicBody.js]: ditto

File: ohara-manager/client/src/api/body/workerBody.js [hunk 4]
```diff
@@ -33,48 +25,10 @@ export const reqConverter = {};
...
-    tags,
-    jarKeys,
-  };
+export const request = params => {
+  const definitions = getCluster(params);
+  const body = createBody(definitions);
+  return body;
 };
 
 export const response = () => {
```
Comment (saivirtue) [inline ohara-manager/client/src/api/body/workerBody.js]: ditto

Comment (wu87988622) [reply ohara-manager/client/src/api/body/workerBody.js]: ok thanks

File: ohara-manager/client/src/api/body/topicBody.js (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/src/api/body/topicBody.js]: ok thanks

File: ohara-manager/client/src/api/body/brokerBody.js (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/src/api/body/brokerBody.js]: ok thanks

File: ohara-manager/client/src/api/body/connectorBody.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/src/api/body/connectorBody.js]: ok thanks

File: ohara-manager/client/src/api/utils/definitionsUtils.js [hunk 5]
```diff
@@ -14,25 +14,108 @@
...
+        obj[key] = obj[key].replace(/\./g, '__');
+      }
     });
-    connectorObj[name] = connect;
+    definitionsObj[obj.key] = obj;
+  });
+  return definitionsObj;
+};
+
+export const getTopic = params => {
```
Comment (konekoya) [inline ohara-manager/client/src/api/utils/definitionsUtils.js]: What does this function do...?

**Comment by wu87988622** [comment]
retry manager-api

**Comment by wu87988622** [comment]
Temporary modification waiting for backend repair #3162

**Comment by wu87988622** [comment]
Temporary modification waiting for backend repair #3162
//cc @oharastream/frontend

**Comment by wu87988622** [comment]
retry manager-api

**Pull request merged**